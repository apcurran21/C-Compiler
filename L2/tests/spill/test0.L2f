(@myF
0

  %myVar <- 0

  rdi <- %myVar

  mem rsp -8 <- :f1_ret
  call @myF2 0
  :f1_ret

  return
)

%myVar

%S
