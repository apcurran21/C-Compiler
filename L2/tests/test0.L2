(@main
  (@main
0
    :call_label0
    goto :call_label1
    :call_label1
    %newVar1 <- 3
    %newVar1 >>= 1
    %newVar0 <- %newVar1
    %newVar0 <- %newVar0
    %newVar0 <<= 1
    %newVar0 <- %newVar0
    %newVar0 += 1
    %newVar0 <- %newVar0
    %newVar0 += 4
    rdi <- %newVar0
    rsi <- 1
    call allocate 2
    %ar <- rax
    %newVar2 <- %ar
    %newVar2 += 8
    mem %newVar2 0  <- 3
    %newVar3 <- %ar
    %newVar3 += 16
    mem %newVar3 0  <- 3
    %newVar5 <- 0
    %newVar6 <- 1
    %newVar5 <- %newVar5
    %newVar5 += 1
    %newVar4 <- %newVar5
    %newVar4 *= 8
    %newVar4 <- %newVar4
    %newVar4 += 24
    %newVar7 <- %ar
    %newVar7 += %newVar4
    mem %newVar7 0  <- 5
    return
  )
)
