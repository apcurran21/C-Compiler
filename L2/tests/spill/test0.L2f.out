(@myF
	0 1
	%S0 <- 0
	mem rsp 0 <- %S0
	%S1 <- mem rsp 0
	rdi <- %S1
	mem rsp -8 <- :f1_ret
	call @myF2 0
	:f1_ret
	return
)
